parmed/openmm/parameters.py

```diff
@@ -122,18 +131,179 @@ def write(self, dest, provenance=None):
+                       'angle="%s" k="%s"/>\n' %
+                       (a1, a2, a3, angle.theteq*tconv, angle.k*kconv))
+        dest.write(' </HarmonicAngleForce>\n')
+
+    def _write_omm_dihedrals(self, dest):
+        if not self.dihedral_types: return
+        # In ParameterSet, dihedral_types is *always* of type DihedralTypeList.
+        # The from_structure method ensures that, even if the containing
+        # Structure has separate dihedral entries for each torsion
+        dest.write(' <PeriodicTorsionForce>\n')
+        diheds_done = set()
+        pconv = u.degree.conversion_factor_to(u.radians)
+        kconv = u.kilocalorie.conversion_factor_to(u.kilojoule)
+        def nowild(name):
+            return name if name != 'X' else ''
+        for (a1, a2, a3, a4), dihed in iteritems(self.dihedral_types):
+            if (a1, a2, a3, a4) in diheds_done: continue
+            diheds_done.add((a1, a2, a3, a4))
+            diheds_done.add((a4, a3, a2, a1))
+            dest.write('  <Proper type1="%s" type2="%s" type3="%s" type4="%s"'
+                       % (nowild(a1), a2, a3, nowild(a4)))
+            for i, term in enumerate(dihed):
+                i += 1
+                dest.write(' periodicity%d="%d" phase%d="%f" k%d="%f"' %
+                           (i, term.per, i, term.phase*pconv, i,
+                            term.phi_k*kconv))
+            dest.write('/>\n')
+        # Now do the periodic impropers. OpenMM expects the central atom to be
+        # listed first. ParameterSet goes out of its way to list it third
+        # (consistent with Amber) except in instances where order is random (as
+        # in CHARMM parameter files). But CHARMM parameter files don't have
+        # periodic impropers, so we don't have to worry about that here.
+        for (a2, a3, a1, a4), improp in iteritems(self.improper_periodic_types):
+            # Try to make the wild-cards in the middle
+            if a4 == 'X':
+                if a2 != 'X':
+                    a2, a4 = a4, a2
+                elif a3 != 'X':
+                    a3, a4 = a4, a3
+            if a2 != 'X' and a3 == 'X':
+                # Single wild-card entries put the wild-card in position 2
+                a2, a3 = a3, a2
+            dest.write('  <Improper type1="%s" type2="%s" type3="%s" '
+                       'type4="%s" periodicity1="%d" phase1="%f" k1="%f"/>\n' %
+                       (a1, nowild(a2), nowild(a3), nowild(a4), improp.per,
+                        improp.phase*pconv, improp.phi_k*kconv)
+            )
+        dest.write(' </PeriodicTorsionForce>\n')
+
+    def _write_omm_impropers(self, dest):
+        if not self.improper_types: return
+        dest.write(' <CustomTorsionForce energy="k*(theta-theta0)^2">\n')
+        dest.write('  <PerTorsionParameter name="k"/>\n')
+        dest.write('  <PerTorsionParameter name="theta0"/>\n')
+        kconv = u.kilocalorie.conversion_factor_to(u.kilojoule)
```

Comment:
Oh, you were suggestion a change.  `parmed.unit` is `simtk.unit` bundled (and the latter replaces this package if it's available).  So `unit` API changes are not up for discussion here (you'd have to take that to pandegroup/openmm, and backwards-incompatible changes in the public-facing API will almost certainly be rejected).

`conversion_factor_to` is better than `conversion_to` or `converted_to` IMO, anyway, since it tells you *exactly* what you get in a nice, concise function name.